src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Emitter.cs

```diff
@@ -156,7 +156,7 @@ private void GenStruct(LoggerMethod lm, string nestedIndentation)
             {nestedIndentation}}}
 ");
                 _builder.Append($@"
-            {nestedIndentation}public static string Format(__{lm.Name}Struct state, global::System.Exception? ex) => state.ToString();
+            {nestedIndentation}public static readonly global::System.Func<__{lm.Name}Struct, global::System.Exception?, string> Format = (state, ex) => state.ToString();
```

Comment:
This is public.  What are the expectations about code accessing this member directly?  If we expect code to access it directly, technically it could be a breaking change, e.g. a consumer doing:
```C#
delegate string MyFunc(Struct state, Exception e);
...
MyFunc f = Format;
```
would previously have succeeded, but now it'll fail to compile.

Reply:
In my opinion, I think we should consider these `__MXStruct` types implementation detail. Note it is a private nested struct in your partial class. So the only way it can be consumed is in your partial class that you are defining your log methods.

Maybe we should add attributes (i.e. EditorBrowsable Never) and comments to these private nested structs instructing developers to not rely on them?

Reply:
That said, @eerhardt, why do we emit these structs and this Format helper then with public visibility?  Why not private?

Reply:
We could try to fix this by always encapsulating our implementation details in _another_ nested class that only has the 1 method visible on it:

```C#
    partial class TestWithDynamicLogLevel 
    {
        private static class __Impl
        {
            private readonly struct __M9Struct : global::System.Collections.Generic.IReadOnlyList<global::System.Collections.Generic.KeyValuePair<string, object?>>
            {
                public static readonly global::System.Func<__M9Struct, global::System.Exception?, string> Format = (state, ex) => state.ToString();
...
            }
            public static void M9(global::Microsoft.Extensions.Logging.LogLevel level, global::Microsoft.Extensions.Logging.ILogger logger)
            {
                if (logger.IsEnabled(level))
                {
                    logger.Log(
                        level,
                        new global::Microsoft.Extensions.Logging.EventId(9, nameof(M9)),
                        new __M9Struct(),
                        null,
                        __M9Struct.Format);
                }
            }
        }

        public static partial void M9(global::Microsoft.Extensions.Logging.LogLevel level, global::Microsoft.Extensions.Logging.ILogger logger)
        {
            __Impl.M9(level, logger);
        }
```

That way there is only 1 method visible to your partial class and everything else is hidden. But I'm not sure it is worth the added complexity.

Reply:
> another example where it'd be really nice to have some mechanism for generating code that can't be referenced by non-generated code.


The half baked idea I had this morning was `file private`. Essentially this type is `private` to the current source file.

Reply:
> The half baked idea I had this morning was file private. Essentially this type is private to the current source file.

Ship it.

Reply:
@stephentoub I could actually work around the BC by caching the static method in a new field. But personally, I agree with @eerhardt that these structs are implementation detail (and not worth the additional complexity), I cannot speak for every dev out there though 😉.

Reply:
I'm fine with whatever call @eerhardt makes here.  I just wanted to highlight that the change could break someone.

Reply:
@maryamariyan - what do you feel about putting `[EditorBrowsable(EditorBrowsableState.Never)]` and a `<summary>` comment on these structs instructing developers to not rely on them?

IMO that would be the best path forward, until we get a compiler feature that would enable this.

Reply:
> @maryamariyan - what do you feel about putting [EditorBrowsable(EditorBrowsableState.Never)] and a <summary> comment on these structs instructing developers to not rely on them?

Sure, that sounds good.

Reply:
@eerhardt @maryamariyan I added the `summary` and `EditorBrowsableState.Never`. I also took the liberty and added an `.editorconfig` which updates `insert_final_newline` for baselines (I found it difficult to edit those in VS), I wasn't quite confident adding it to a global `.editorconfig` but I'd be happy to do so 😉